### ISSUE TITLE
seat position may change after player motion

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#1058]
Preserve the seat position after tables are combined and when players move.

A player reported:

> Another thing – UI seems to get confused on game change. I rotated to the top position at the table, and when the game switched to Razz it moved me back.  Will let you know if it does it all the time.

I asked in email:

> Was it just a switch to Razz, or was it when the tables were combined? I've seen a problem with table combinations.

and haven't yet seen a response. 

I've definitely seen situations where my position at the table changes after some sort of player motion. I haven't written them down (AFAIK) and don't know if it can happen with rebalancing or table combining or both. It shouldn't be too hard to look at the code to see what the client does in both of those cases. My guess is at least one of them has a fairly obviously missing piece of logic or fairly obviously incorrect logic.

Since this has happened to me and I never made an issue out of it, my guess is it's happening to others and only our intrepid reporter decided to take his time to report it. Thanks!

**Comment by ctm** [comment]
BTW, the original report does not mention player motion, but I'm pretty skeptical that player motion wasn't involved, because game switches are super common and I can't think of anything in the code that would cause that problem on game switches whereas there are obvious places where I could have messed up player motion.

**Comment by ctm** [comment]
This just happened to me in event 3781 after table combination.

**Comment by ctm** [comment]
I'm going to take a stab at this "soon" (probably today). I still have a lot of business stuff to do, but it's a long weekend and I'll feel better if I can fix a bug or two and add a feature or two.

**Comment by ctm** [comment]
Looking closer at the code, the receipt of any `Status` message calls `update_rotation`. Since we receive a `Status` after player motion, I'll need to figure out why `update_rotation` isn't doing the right thing. Unfortunately, although I can indeed set up a tournament with 2-seated tables and 3 players, the two times I've tried to do it, the proper thing happened after the table merge.

As such, this issue is no longer `easy`, thus I need to strip it of `high priority` so I don't waste time on something that doesn't come up much and when it does can be ameliorated by manual rotation.

Saddenz.

**Comment by ctm** [comment]
I continued working on this, because I saw there was some easy stuff I could still do. I did that stuff and it mostly worked, but I was still seeing some weirdness. After adding a ton of `log::info!` invocations it looks very much like the `seat` field of the `PlayerInfo` isn't being updated after a table merge, which is not something I had considered.

If I'm right, then fixing that will allow me to close this issue, because the other work I already did fixes the other half of the bug. I don't think I'll get it done in time for this evening's game though.

**Comment by ctm** [comment]
Fixed. Deploying now.

**Comment by ctm** [comment]
Oh, and for the record, it wasn't that the `seat` field was incorrect. It as we weren't resetting rotation to 0 if our desired seat matched the seat we got. So, if our rotation was 3, because we started in 0, but wanted seat 3, then later we got placed in seat 3, we'd see that we wanted 3 and got 3, so there's nothing left to do. Well, that was wrong. The thing to do there is to set rotation back to 0.

I'm glad I didn't put this on the back burner when I stripped the `high priority` label, because there were a few different layers that needed to be touched and it took a little while to reload my synapses.

On one hand, I didn't get anything else done on mb2 today, but on the other hand, this one was trickier than it appeared.
